From 703d379a09ffcc5e2ab394a667904cdc9fd177c3 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:18:28 -0400
Subject: [fs] nfsd: Remove iattr parameter from nfsd_symlink()

Message-id: <1410898849-22126-25-git-send-email-bfields@redhat.com>
Patchwork-id: 95848
O-Subject: [Patch RHEL7 024/165] NFSD: Remove iattr parameter from nfsd_symlink()
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Kinglong Mee <kinglongmee@gmail.com>

Commit db2e747b1499 (vfs: remove mode parameter from vfs_symlink())
have remove mode parameter from vfs_symlink.
So that, iattr isn't needed by nfsd_symlink now, just remove it.

Signed-off-by: Kinglong Mee <kinglongmee@gmail.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 1e444f5bc0c468e244ee601b7acbd87f0b6ee7e2
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs3proc.c b/fs/nfsd/nfs3proc.c
index 19ba233..fa2525b 100644
--- a/fs/nfsd/nfs3proc.c
+++ b/fs/nfsd/nfs3proc.c
@@ -282,8 +282,7 @@ nfsd3_proc_symlink(struct svc_rqst *rqstp, struct nfsd3_symlinkargs *argp,
  fh_copy(&resp->dirfh, &argp->ffh);
  fh_init(&resp->fh, NFS3_FHSIZE);
  nfserr = nfsd_symlink(rqstp, &resp->dirfh, argp->fname, argp->flen,
-         argp->tname,
-         &resp->fh, &argp->attrs);
+         argp->tname, &resp->fh);
  RETURN_STATUS(nfserr);
 }
 
diff --git a/fs/nfsd/nfs4proc.c b/fs/nfsd/nfs4proc.c
index 552cc03..673eb4b 100644
--- a/fs/nfsd/nfs4proc.c
+++ b/fs/nfsd/nfs4proc.c
@@ -622,8 +622,7 @@ nfsd4_create(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
  case NF4LNK:
   status = nfsd_symlink(rqstp, &cstate->current_fh,
           create->cr_name, create->cr_namelen,
-          create->cr_data,
-          &resfh, &create->cr_iattr);
+          create->cr_data, &resfh);
   break;
 
  case NF4BLK:
diff --git a/fs/nfsd/nfsproc.c b/fs/nfsd/nfsproc.c
index 583ed03..eff4955 100644
--- a/fs/nfsd/nfsproc.c
+++ b/fs/nfsd/nfsproc.c
@@ -409,9 +409,7 @@ nfsd_proc_symlink(struct svc_rqst *rqstp, struct nfsd_symlinkargs *argp,
   */
  argp->tname[argp->tlen] = '\0';
  nfserr = nfsd_symlink(rqstp, &argp->ffh, argp->fname, argp->flen,
-       argp->tname,
-        &newfh, &argp->attrs);
-
+       argp->tname, &newfh);
 
  fh_put(&argp->ffh);
  fh_put(&newfh);
diff --git a/fs/nfsd/vfs.c b/fs/nfsd/vfs.c
index 552a61a..edae097 100644
--- a/fs/nfsd/vfs.c
+++ b/fs/nfsd/vfs.c
@@ -1656,8 +1656,7 @@ __be32
 nfsd_symlink(struct svc_rqst *rqstp, struct svc_fh *fhp,
     char *fname, int flen,
     char *path,
-    struct svc_fh *resfhp,
-    struct iattr *iap)
+    struct svc_fh *resfhp)
 {
  struct dentry *dentry, *dnew;
  __be32  err, cerr;
diff --git a/fs/nfsd/vfs.h b/fs/nfsd/vfs.h
index 36bc2e0..3aeb346 100644
--- a/fs/nfsd/vfs.h
+++ b/fs/nfsd/vfs.h
@@ -89,7 +89,7 @@ __be32  nfsd_readlink(struct svc_rqst *, struct svc_fh *,
     char *, int *);
 __be32  nfsd_symlink(struct svc_rqst *, struct svc_fh *,
     char *name, int len, char *path,
-    struct svc_fh *res, struct iattr *);
+    struct svc_fh *res);
 __be32  nfsd_link(struct svc_rqst *, struct svc_fh *,
     char *, int, struct svc_fh *);
 __be32  nfsd_rename(struct svc_rqst *,
-- 
1.7.1